modules/@apostrophecms/piece-type/index.js

```diff
@@ -365,20 +366,27 @@ module.exports = {
       composeFilters() {
         self.filters = Object.keys(self.filters).map(key => ({
           name: key,
-          ...self.filters[key]
+          ...self.filters[key],
+          inputType: self.filters[key].inputType || 'select'
         }));
         // Add a null choice if not already added or set to `required`
         self.filters.forEach(filter => {
-          if (
-            !filter.required &&
-            filter.choices &&
-            !filter.choices.find(choice => choice.value === 'any')
-          ) {
-            filter.def = 'any';
-            filter.choices.push({
-              value: 'any',
-              label: 'None'
-            });
+          if (filter.choices) {
+            if (
+              !filter.required &&
+              filter.choices &&
+              !filter.choices.find(choice => choice.value === null)
+            ) {
+              filter.def = null;
+              filter.choices.push({
+                value: null,
+                label: 'None'
+              });
+            }
+          } else {
+            // Dynamic choices from the REST API, but
+            // we need a label for "no opinion"
+            filter.nullLabel = 'Choose One';
```

Comment:
This is only in play when there are no set choices? I'm not clear on why "choose one" makes sense then.

Reply:
The REST API will supply the choices, but it doesn't supply an "I don't care" choice, so we must supply that. "Choose One" is a good label because it is then understood that you can start filtering by changing it.